src/client/auth/scram.rs

```diff
@@ -258,8 +258,13 @@ impl ScramVersion {
 
         if should_update_cache {
             if let Ok(ref mut cache) = CREDENTIAL_CACHE.write() {
-                if cache.get(&cache_entry_key).is_none() {
-                    cache.insert(cache_entry_key, salted_password);
+                match cache.entry(cache_entry_key) {
```

Comment:
I think this can be simplified by using `if let` to avoid defining an empty branch:

```rust
if let Entry::Vacant(entry) = cache.entry(cache_entry_key) {
    entry.insert(salted_password);
}
```

Reply:
I'm sorry this fell under the rag. I've pushed an update.

Reply:
No worries! Thanks for making the change